### PR TITLE
Add-PoshGitToProfile should not modify signed prof

### PR DESCRIPTION
**Pull request by rkeithhill**
Forgot to address this issue before shipping 0.7.0 - https://github.com/dahlbyk/posh-git/issues/328#issuecomment-269740506

Merge to develop.

### CODE REVIEW

File: src/Utils.ps1 [hunk 1]
```diff
@@ -138,6 +152,16 @@ function Add-PoshGitToProfile([switch]$AllHosts, [switch]$Force, [switch]$WhatIf
         return
     }
 
+    # If the profile script exists and is signed, then we should not modify it
+    if (Test-Path -LiteralPath $profilePath) {
+        $sig = Get-AuthenticodeSignature $profilePath
+        if ($sig.Status -eq [System.Management.Automation.SignatureStatus]::Valid) {
```
Comment (dahlbyk) [inline src/Utils.ps1]: Should we warn/abort if `$sig.Status` is anything other than `NotSigned`?

File: src/Utils.ps1 [hunk 2]
```diff
@@ -82,18 +82,32 @@ function Invoke-Utf8ConsoleCommand([ScriptBlock]$cmd) {
 .OUTPUTS
     None.
 #>
-function Add-PoshGitToProfile([switch]$AllHosts, [switch]$Force, [switch]$WhatIf) {
+function Add-PoshGitToProfile {
+    [CmdletBinding(SupportsShouldProcess)]
```
Comment (dahlbyk) [inline src/Utils.ps1]: 👍 Always forget about this.

File: src/Utils.ps1 (hunk 1 above)
Comment (rkeithhill) [inline src/Utils.ps1]: Yup. I'll make that change.

Comment (rkeithhill) [inline src/Utils.ps1]: Well that turned out to be a little ugly as there are three values that can represent an unsigned file (NotValid, UnknownError, IncompatibleFileType) and four values that can represent a signed file.  Switching to testing the SignatureType field.

**Pull request merged**